scout/server/blueprints/variants/controllers.py

```diff
@@ -1488,7 +1489,38 @@ def persistent_filter_actions(
         filter_id = request_form.get("filters")
         filter_obj = store.retrieve_filter(filter_id)
         if filter_obj is not None:
-            form = FiltersFormClass(MultiDict(filter_obj))
+
+            def _coerce_formdata(filter: dict) -> MultiDict:
+                """According to wtforms docs https://wtforms.readthedocs.io/en/3.1.x/forms/#the-form-class,
+                formdata passed to Form init needs to be pre-coerced to types. Hence especially the replace
+                of locale specific chars in numbers on class process_formdata would not be enough for the
+                better_decimal form fields.
+                """
+                for filter_decimal_value_form_field in [
+                    "cadd_score",
+                    "revel",
+                    "gnomad_frequency",
+                    "control_frequency",
+                    "tumor_frequency",
+                ]:
+                    if (
+                        filter_decimal_value_form_field in filter
+                        and type(filter[filter_decimal_value_form_field][0]) is str
+                    ):
+                        try:
+                            raw_value = decimal.Decimal(
+                                filter[filter_decimal_value_form_field][0].replace(",", ".")
+                            )
+                        except (decimal.InvalidOperation, ValueError):
+                            raw_value = filter[filter_decimal_value_form_field][0]
+                            raise ValueError("Not a valid decimal value")
```

Comment:
Won't this make the page crash?

Reply:
I didnt think so - we used to catch them, but now I moved it up slightly to before the form cerationg, so could be that the catching happens in the WTForm. I'll modify - doesn't hurt!